code/modules/mob/mob_helpers.dm

```diff
@@ -317,7 +317,11 @@
 /mob/proc/hallucinating()
 	return FALSE
 
-
+/mob/living/carbon/hallucinating()
+	if(has_status_effect(/datum/status_effect/hallucination))
+		return TRUE
+	return FALSE
```

Comment:
just remove the proc above and change all checks to has status effects, i didn't do it originally cause the pr was already too big

Reply:
ok